src/RoslynAnalyzerServer/Analysis/Configuration/RoslynAnalysisConfigurationProvider.cs

```diff
@@ -32,16 +33,45 @@ internal class RoslynAnalysisConfigurationProvider(
     ISonarLintXmlProvider sonarLintXmlProvider,
     IRoslynAnalyzerProvider roslynAnalyzerProvider,
     IRoslynAnalysisProfilesProvider analyzerProfilesProvider,
+    IAsyncLockFactory asyncLockFactory,
+    IThreadHandling threadHandling,
     ILogger logger) : IRoslynAnalysisConfigurationProvider
 {
     private readonly ILogger logger = logger.ForContext(Resources.RoslynAnalysisLogContext, Resources.RoslynAnalysisConfigurationLogContext);
 
-    public IReadOnlyDictionary<Language, RoslynAnalysisConfiguration> GetConfiguration(List<ActiveRuleDto> activeRules, Dictionary<string, string>? analysisProperties, AnalyzerInfoDto analyzerInfo)
-    {
-        // todo add caching https://sonarsource.atlassian.net/browse/SLVS-2481
+    private readonly IAsyncLock asyncLock = asyncLockFactory.Create();
+    private IReadOnlyDictionary<Language, RoslynAnalysisConfiguration>? cachedConfigurations;
```

Comment:
to reduce the chance of 2 cache fields modified independently, add a private readonly struct to hold the pair together. For example, https://github.com/SonarSource/sonarlint-visualstudio/blob/2b25230bd84473cde42a2a793a6533401e4e8a45/src/Core/GitWorkSpaceService.cs#L47